src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/utils/SPARQL2GraphQLHelper.java

```diff
@@ -0,0 +1,249 @@
+            final Node nestedSubgraphNode, final String currentPath, final String sgpType, final String predicateURI){
+
+        final String alias = config.removePropertyPrefix(predicateURI);
+        final String fieldName = config.removePropertySuffix(alias);
+        final String newPath = GraphQLFieldPathBuilder.addObject(currentPath, alias, fieldName);
+        final String nestedType = endpoint.getGraphQLFieldValueType(sgpType, fieldName);
+
+        addSgp(fieldPaths,subgraphPatterns,connectors,config,endpoint,
+            nestedSubgraphNode,newPath,nestedType);
+    }
+
+    /**
+     * Helper function to add a field to the query that represents a nested object that only needs an id field (no more recursive calls)
+     */
+    public static void addEmptyObjectField(final Set<String> fieldPaths, final GraphQL2RDFConfiguration config,
+            final GraphQLEndpoint endpoint, final String currentPath, final String sgpType, 
+            final String predicateURI){
+        final String alias = config.removePropertyPrefix(predicateURI);
+        final String fieldName = config.removePropertySuffix(alias);
+        final String newPath = GraphQLFieldPathBuilder.addObject(currentPath, alias, fieldName);
+        final String nestedType = endpoint.getGraphQLFieldValueType(sgpType, fieldName);
+        fieldPaths.add(GraphQLFieldPathBuilder.addID(newPath, nestedType));
+    }
+
+    /**
+     * Helper function to add a field to the query that represents a scalar value
+     */
+    public static void addScalarField(final Set<String> fieldPaths, final GraphQL2RDFConfiguration config, 
+            final String currentPath, final String predicateURI){
+        final String alias = config.removePropertyPrefix(predicateURI);
+        final String fieldName = config.removePropertySuffix(alias);
+        fieldPaths.add(GraphQLFieldPathBuilder.addScalar(currentPath, alias, fieldName));
+    }
+
+
+
+    /**
+     * Helper function which checks if any triple pattern predicate in @param sgp is a variable or blank node, returns true if so
+     */
+    public static boolean hasVariablePredicate(final Set<TP> sgp){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            if(predicate.isVariable()){
+                return true;
+            }    
+        }
+        return false;
+    }
+
+    /**
+     * Returns the GraphQL object type @param sgp correponds to. 
+     * If the type is undeterminable @return null
+     */
+    public static String determineSgpType(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
```

Comment:
the second condition should be a method of the Configuration interface, that receives a string and says if it is a possible property (is that the correct GraphQL nomenclature?) or not.